homeassistant/components/glinet/device_tracker.py

```diff
@@ -0,0 +1,138 @@
+"""Support for GLinet routers."""
+from __future__ import annotations
+
+from typing import Any
+
+from homeassistant.components.device_tracker import SOURCE_TYPE_ROUTER
+from homeassistant.components.device_tracker.config_entry import ScannerEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.device_registry import CONNECTION_NETWORK_MAC
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+from homeassistant.helpers.entity import DeviceInfo
+
+from .const import DATA_GLINET, DOMAIN
+from .router import ClientDevInfo, GLinetRouter
+
+DEFAULT_DEVICE_NAME = "Unknown device"
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities
+) -> None:
+    """Set up device tracker for GLinet component."""
+    router: GLinetRouter = hass.data[DOMAIN][entry.entry_id][DATA_GLINET]
+    tracked: set[str] = set()
+
+    @callback
+    def update_router():
+        """Update the values of the router."""
+        add_entities(router, async_add_entities, tracked)
+
+    router.async_on_close(
+        async_dispatcher_connect(hass, router.signal_device_new, update_router)
+    )
+
+    update_router()
+
+
+@callback
+def add_entities(router: GLinetRouter, async_add_entities, tracked):
+    """Add new tracker entities from the router."""
+    new_tracked = []
+
+    for mac, device in router.devices.items():
+        if mac in tracked:
+            continue
+
+        new_tracked.append(GLinetDevice(router, device))
+        tracked.add(mac)
+
+    if new_tracked:
+        async_add_entities(new_tracked)
+
+
+class GLinetDevice(ScannerEntity):
+    """Representation of a GLinet device."""
+
+    def __init__(self, router: GLinetRouter, device: ClientDevInfo) -> None:
+        """Initialize a GLinet device."""
+        self._router = router
+        self._device = device
+
+    @property
+    def unique_id(self) -> str:
+        """Return a unique ID."""
+        return self._device.mac
+
+    @property
+    def name(self) -> str:
+        """Return the name."""
+        return self._device.name or DEFAULT_DEVICE_NAME
+
+    @property
+    def is_connected(self):
+        """Return true if the device is connected to the network."""
+        return self._device.is_connected
+
+    @property
+    def source_type(self) -> str:
+        """Return the source type."""
+        return SOURCE_TYPE_ROUTER
+
+    @property
+    def extra_state_attributes(self) -> dict[str, Any]:
+        """Return the attributes."""
+        attrs = {}
+        if self._device.last_activity:
+            attrs["last_time_reachable"] = self._device.last_activity.isoformat(
```

Comment:
This means that device states are constantly updated. We shouldn't include this in the device tracker entity. It can be a sensor, but added in a future PR and disabled by default.